NewHorizons/Builder/Props/DetailBuilder.cs

```diff
@@ -106,6 +105,13 @@ public static GameObject Make(GameObject go, Sector sector, GameObject prefab, P
 
                 foreach (var component in prop.GetComponentsInChildren<Component>(true))
                 {
+                    // Components can come through as null here (yes, really),
+                    // Usually if a script was added to a prefab in an asset bundle but isn't present in the loaded mod DLLs
+                    if (component == null)
+                    {
+                        invalidComponentFound = true;
```

Comment:
Why don't we log the error here instead of looping again later?

Reply:
This loop doesn't have the context of which GameObject the bad component is attached to, since `component.gameObject` throws a NullReferenceException, and being able to give the user the path to the invalid component is very helpful. I could merge the two loops by having an outer loop through all Transforms and then an inner loop through all components on that Transform's game object, but that would introduce a lot more overhead for all the different GetComponents calls and I was worried about slowing down DetailBuilder even more than it already is.

Reply:
Oh okay, that makes sense then